### DISCUSSION TITLE
ABACUS v3.3 is scheduled to be released on July 1, 2023.

### DISCUSSION THREAD

**Discussion by pxlxingliang** [deepmodeling/abacus-develop#2500] [General]
ABACUS v3.3 is scheduled to be released on July 1, 2023. We plan to offer the following improvements:
## Features
- The interface with ShengBTE, a software package for solving the Boltzmann Transport Equation for phonons and computing the lattice contribution to the thermal conductivity of bulk crystalline solids and nanowires.
- Add the using of LibRI hybrid functional to do the calculation of realx and MD.
- Reconstruct Hamiltonian in the LCAO basis into new HContainer objects, which can facilitate high-performance optimization, new algorithm development, testing, and other related work.
- Add dump interface, which is a unified interface for outputting data during the calculation, and can improve debug efficiency.
- A more robust LCAO basis processing method which facilitates convergence with respect to cutoff energies and is less sensitive to orbital qualities.

## Coding Standard
- Add code style check for new PR
- Add coverage ratio check for new PR

## Autotest Workflow and Code Coverage
- Use the abacustest application on the launching platform to regularly perform some long-term testing tasks to monitor the performance and functional stability of ABACUS.
- In this version, code coverage will reach 80%.

**Comment by Satinelamp** [comment]
What is a HContainer object?

**Reply by pxlxingliang** [reply]
> What is a HContainer object?

@dyzheng Could you help supply some introduction?

**Comment by Satinelamp** [comment]
Would ABACUS 3.3 release the TDDFT function officially?

**Comment by mohanchen** [comment]
depends on the progress
